crates/cargo-util-schemas/src/core/package_id_spec.rs

```diff
@@ -11,6 +9,8 @@ use crate::core::PartialVersion;
 use crate::core::SourceKind;
 use crate::manifest::PackageName;
 
+type Result<T> = std::result::Result<T, PackageIdSpecError>;
```

Comment:
(github is not letting me highlight `PackageIdSpec::parse` and `from_url`)

Looks like we are exposing `PackageIdSpecError` in the crate's public API but not exposing the type.